spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -506,7 +512,7 @@ public final void start() {
 		checkGroupId();
 		this.lifecycleLock.lock();
 		try {
-			if (!isRunning()) {
+			if (!isRunning() && this.isContainerFenced()) {
 				Assert.state(this.containerProperties.getMessageListener() instanceof GenericMessageListener,
```

Comment:
I would just add here the line of code instead:
```
Assert.state(!this.fenced, "Container Fenced. It is not allowed to start.");
```

Otherwise we may debate if the name of that new method is correct and its usage reflect the purpose.

See `Assert.state()` implementation for details.